lesson09.md

```diff
@@ -0,0 +1,216 @@
+<!-- .slide: id="lesson9" -->
+
+# Basic Frontend - Spring 2021
+
+Lesson 9, Thursday, 2021-04-01
+
+---
+
+### Warmup quizzes :)
+
+What does this program output to console?
+
+```js
+function sayHello() {
+    console.log("Hello!");
+}
+```
+
+Nothing. We never call the function!<!-- .element: class="fragment" -->
+
+---
+
+```js
+function sayHello(name) {
+    let result = "Hello " + name;
+}
+
+console.log(sayHello("Harald"));
+```
+
+`undefined`<!-- .element: class="fragment" -->
+
+---
+
+Could you think of a better name for `foo`?
+
+```js
+function foo(a, b) {
+    if (a < b) {
+        return a;
+    } else {
+        return b;
+    }
+}
+
+let result = foo(4, 5);
+```
+
+`minimum`<!-- .element: class="fragment" -->
+
+---
+
+### Solutions from lesson 8
+
+Create a function that calculates the `square` of a number:
+
+```js
+function square(num) {
+    return num ** 2; // or: num * num
+}
+```
+
+---
+
+Write a function called `celsiusToFahrenheit`:
+
+```js
+function celsiusToFahrenheit(degreesInCelsius) {
+    let degreesInFahrenheit = degreesInCelsius * 1.8 + 32;
+    return degreesInFahrenheit;
+}
+```
+
+---
+
+Create a function that takes a day of the week and returns a message depending on the day:
+
+```js
+function messageOfTheDay(day) {
+    if (day === "Monday") {
+        return "Sucks";
+    }
+    if (day === "Friday") {
+        return "Almost Weekend!";
+    }
+    if (day === "Saturday" || day === "Sunday") {
+        return "Yippee!!!";
+    }
+    return "It's OK.";
+}
+```
+
+---
+
+### Variable lifetime
+
+```js
+let greeting = "Hello";
+
+if (greeting === "Hello") {
+    let name = "Sevtap";
+    console.log(greeting + " " + name);s
+}
+```
+
+What's the lifetime of `greeting` and `name`?
+
+---
+
+* Variables defined inside a code block `{}` exist until the end of the code block
+* Variables defined in the global scope (e.g. outside of code blocks) exist until the user closes the page.
+
+---
+
+### HTML and JavaScript
+
+We've learned that we can load a JavaScript file from within HTML:
+
+```html
+<script src="main.js"></script>
+```
+
+And we can use `console.log` to output values to the debug console.
+
+---
+
+Let's try to actually change something in our web page. The browser defined a lot of variables for us to play with, e.g.:
+
+```js
+// set the document's background color by assigning
+// a color to this variable:
+document.body.style.backgroundColor = "red";
+```
+
+You can choose your favorite color from [https://developer.mozilla.org/en-US/docs/Web/CSS/color_value](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value)
+
+---
+
+We can also change the font size, e.g.:
+
+```js
+document.body.style.fontSize = "1.5em";
+```
+
+(`em` is a unit in typography, see [wikipedia](https://en.wikipedia.org/wiki/Em_(typography)))
```

Comment:
I'm wondering if it's not better to link an MDN page where they talk about the`em` unit in the context of CSS. Even though the name em comes from there, the behaviour in CSS is a bit different than the one in real typography